dialogue-core/src/main/java/com/palantir/dialogue/core/UserAgentEndpointChannel.java

```diff
@@ -56,8 +56,19 @@ static EndpointChannel create(EndpointChannel delegate, Endpoint endpoint, UserA
     }
 
     private static UserAgent augmentUserAgent(UserAgent baseAgent, Endpoint endpoint) {
+        String endpointVersion = endpoint.version();
+
+        // Until conjure-java 5.14.2, we mistakenly embedded 0.0.0 in everything. This fallback logic attempts
+        // to work-around this and produce a more helpful user agent
+        if (endpointVersion.equals("0.0.0")) {
```

Comment:
I wonder if we could make the endpoint version be optional and then exclude it from code gen by default. It would allow us to cache conjure generation much more aggressively rather than force regen on each commit/version

Reply:
I think we codegen the impl version lookip

Reply:
Yep, see https://github.com/palantir/conjure-java/pull/879/files#diff-9fd8c6f3733ade117b02507eef44be37R45-R47